images/keycloak/tests/01-docker-production-mode.sh

```diff
@@ -0,0 +1,175 @@
+  --https-key-store-password=$KEYSTORE_PASSWORD
+}
+
+search_logs() {
+  local retries=6
+  local delay=15 # seconds
+
+  for ((i=1; i<=retries; i++)); do
+    local logs=$(docker logs "${container_id}" 2>&1)
+    local all_terms_found=true
+
+    for term in "${terms[@]}"; do
+      if echo "$logs" | grep -Fq "$term"; then
+        echo "Found log term: $term"
+      else
+        echo "Log term NOT found: $term (attempt $i of $retries)"
+        all_terms_found=false
+      fi
+    done
+
+    if $all_terms_found; then
+      return 0
+    elif [[ $i -lt $retries ]]; then
+      echo "Some log terms were missing. Retrying in $delay seconds..."
+      sleep $delay
+    fi
+  done
+
+  # After all retries, record the missing terms
+  for term in "${terms[@]}"; do
+    if ! echo "$logs" | grep -Fq "$term"; then
+      missing_terms+=("$term")
+    fi
+  done
+
+  echo "FAILED: After $retries attempts, the following terms were not found:"
+  printf '%s\n' "${missing_terms[@]}"
+  exit 1
+}
+
+
+keycloak_api_get_users() {
+    # Get Keycloak access token
+    local response=$(curl -k -s -w "\nHTTP_STATUS_CODE:%{http_code}\n" -X POST $KEYCLOAK_URL/realms/master/protocol/openid-connect/token \
+        --user admin-cli:Psip5UvTO1EXUEwzb15nxLWnwdU1Nlcg \
+        -H 'content-type: application/x-www-form-urlencoded' \
+        -d "username=admin&password=$KEYSTORE_PASSWORD&grant_type=password")
+
+    local response_code=$(echo "$response" | grep "HTTP_STATUS_CODE:" | cut -d':' -f2)
+    local content=$(echo "$response" | sed '/HTTP_STATUS_CODE/d')
+
+    if [[ "$response_code" != "200" ]]; then
+        echo "FAILED: HTTP response code is $response_code"
+        exit 1
+    fi
+
+    local access_token=$(echo "$content" | jq --raw-output '.access_token')
+
+    # Get list of users that exist in Keycloak
+    curl -k -X GET $KEYCLOAK_URL/admin/realms/master/users -H "Authorization: Bearer $access_token" | jq
+}
+
+
+TEST_container_starts_ok() {
+    # Create Keystore and launch Keycloak
+    create_keystore
+    local -r container_id=$(start_container)
+    trap "docker stop ${container_id} && rm -rf ${KEYSTORE_PATH}" EXIT
+
+    # Check if the container is running
+    sleep 15
+    if ! docker ps --filter "name=local-keycloak" --format '{{.Names}}' | grep -q "^local-keycloak$"; then
+        echo "FAILED: Container local-keycloak is not running."
+        exit 1
+    else
+        echo "Container local-keycloak is running."
+    fi
+
+    # Look for each log term. Will record any which are not found.
+    search_logs
+
+    if [[ ${#missing_terms[@]} -ne 0 ]]; then
+        echo "The following terms were not found:"
+        printf '%s\n' "${missing_terms[@]}"
+        exit 1
+    fi
+}
+
+TEST_keycloak_api_accessible() {
+  local retries=3
+  local delay=5 # seconds
+  local attempt=0
+
+  while [[ $attempt -le $retries ]]; do
+    local response_code=$(curl -k -I -s -o /dev/null -w "%{http_code}" "$KEYCLOAK_URL/realms/master")
```

Comment:
Does it make sense to combine `curl` flags like so?

```suggestion
    local response_code=$(curl -kIso /dev/null -w "%{http_code}" "$KEYCLOAK_URL/realms/master")
```